pkg/controllers/user/pipeline/engine/engine.go

```diff
@@ -17,19 +19,29 @@ type PipelineEngine interface {
 
 func New(cluster *config.UserContext) PipelineEngine {
 
-	nodeLister := cluster.Core.Nodes("").Controller().Lister()
 	serviceLister := cluster.Core.Services("").Controller().Lister()
 	secrets := cluster.Core.Secrets("")
 	secretLister := secrets.Controller().Lister()
 	managementSecretLister := cluster.Management.Core.Secrets("").Controller().Lister()
 	sourceCodeCredentialLister := cluster.Management.Management.SourceCodeCredentials("").Controller().Lister()
+
+	dial, err := cluster.Management.Dialer.ClusterDialer(cluster.ClusterName)
+	if err != nil {
+		logrus.Warnf("Failed to get cluster dialer: %v", err)
```

Comment:
hm, and we just proceed if the error happens, and set the dial to nil on line 34? Is it expected behavior?

Reply:
Looks like clusterDialer can be a nodeDialer and may be invalid over time.  I think we should get the dial before doing the HTTP connection, not on controller registration. And we can check and handle the error there

Reply:
updated